lib/set_game/game_server.ex

```diff
@@ -1,19 +1,53 @@
 defmodule SetGame.GameServer do
-  use GenServer
-
+  use GenServer, start: {__MODULE__, :start_link, []}, restart: :transient
   alias SetGame.Board
   alias SetGame.Player
 
   defmodule State do
-    defstruct board: Board.new(), players: [], state: :players_joining
+    defstruct board: Board.new(),
+              name: "",
+              players: [],
+              state: :players_joining
   end
 
   # Client
 
-  def start_link(_arg \\ nil) do
-    GenServer.start_link(__MODULE__, %State{})
+  @spec start_link(any()) :: :ignore | {:error, any} | {:ok, pid}
+  def start_link({:via, _, {_, name}} = via_tuple_name \\ via_tuple(find_available_name())) do
+    GenServer.start_link(__MODULE__, fresh_state(name), name: via_tuple_name)
   end
 
+  defp timeout(), do: Application.get_env(:set_game, :timeout)
+
+  defp fresh_state(name) do
+    %State{name: name}
+  end
+
+  defp generate_name() do
+    symbols = '23456789ABCDEFGHJKMNPQRSTUWXYZ'
```

Comment:
If these symbols are constant you should move them into module attributes along with the count. Even though it is small you can avoid the overhead on subsequent function calls:

```elixir
@symbols '23456789ABCDEFGHJKMNPQRSTUWXYZ'
@symbol_count length(@symbols)

defp generate_name() do
  for _ <- 1..10, into: "", do: <<Enum.at(@symbols, floor(:rand.uniform() * @symbol_count))>>
end
```

Reply:
```elixir
@symbols |> Enum.take_random(10) |> :binary.list_to_bin()
```

Reply:
@schrockwell the problem with `Enum.take_random` is that it cannot get repeated elements in there:

```elixir
 def tr(list \\ Enum.to_list(1..10), max_tries \\ 1_000_000, tries_count \\ 0, success \\ 0) do
    desired_length = length(list) - 1
    rand_coll = Enum.take_random(list, desired_length)
    has_repeated_element = length(Enum.uniq(rand_coll)) < desired_length

    cond do
      tries_count == max_tries ->
        IO.puts("Only #{success} has repeated elements out of #{tries_count} tries.")

      true ->
        tr(
          list,
          max_tries,
          tries_count + 1,
          if has_repeated_element do
            success + 1
          else
            success
          end
        )
    end
  end
```